core/services/chainlink/config_test.go

```diff
@@ -1050,18 +1041,60 @@ func TestNewGeneralConfig_SecretsOverrides(t *testing.T) {
 	assert.Equal(t, DBURL_OVERRIDE, (&dbURL).String())
 }
 
-//go:embed testdata/secrets-partial.toml
-var invalidSecretsTOML string
-
 func TestSecrets_Validate(t *testing.T) {
-	var invalid Secrets
-	d := toml.NewDecoder(strings.NewReader(invalidSecretsTOML)).DisallowUnknownFields()
-	require.NoError(t, d.Decode(&invalid))
+	for _, tt := range []struct {
+		name string
+		toml string
+		exp  string
+	}{
+		{name: "partial",
+			toml: `ExplorerAccessKey = "access_key"
+ExplorerSecret = "secret"`,
+			exp: `2 errors:
+	1) DatabaseURL: empty: must be provided and non-empty
+	2) KeystorePassword: empty: must be provided and non-empty`},
+
+		{name: "invalid-urls",
+			toml: `DatabaseURL = "postgresql://user:passlocalhost:5432/asdf"
+DatabaseBackupURL = "foo-bar?password=asdf"`,
+			exp: `3 errors:
+	1) DatabaseURL: invalid value *****: missing or insufficiently complex password: DB URL must be authenticated; plaintext URLs are not allowed. Database should be secured by a password matching the following complexity requirements: 
+Must have a length of 16-50 characters
+Must not comprise:
+	Leading or trailing whitespace (note that a trailing newline in the password file, if present, will be ignored)
+
+	2) DatabaseBackupURL: invalid value *****: missing or insufficiently complex password: 
+Expected password complexity:
+Must be at least 16 characters long
+Must not comprise:
+	Leading or trailing whitespace
+	A user's API email
+
+Faults:
+	password is 4 characters long
```

Comment:
This probably reveals too much information. I don't think it was introduced by your change here but we should address it anyway.

Reply:
In this PR? Do you have any suggestions? If not, can it be a follow up story?

Reply:
Will follow up in a separate PR